pkg/pillar/zfs/zfs.go

```diff
@@ -45,6 +46,36 @@ func MountDataset(log *base.LogObject, dataset string) (string, error) {
 	return string(stdoutStderr), err
 }
 
+// GetZfsStatusStr returns detailed status of pool
+func GetZfsStatusStr(log *base.LogObject, pool string) string {
+	args := append(zpoolPath, "status", pool)
+	stdoutStderr, err := base.Exec(log, vault.ZfsPath, args...).CombinedOutput()
+	if err != nil {
+		log.Errorf("zpool status error: %s", err)
+		return ""
+	}
+	var status []string
+	inStatus := false
+	scanner := bufio.NewScanner(strings.NewReader(string(stdoutStderr)))
+	for scanner.Scan() {
+		text := strings.TrimSpace(scanner.Text())
+		// we expect 'status:' in the beginning to start capture output
+		if strings.HasPrefix(strings.TrimSpace(text), "status:") {
+			inStatus = true
+			text = strings.TrimPrefix(text, "status:")
+		} else
+		// status ends with 'action:' or 'config:' in the beginning of the line
+		if strings.HasPrefix(text, "action:") ||
```

Comment:
Do we want to send the action in future release ? May be once we have UI workflow going we can display action too.

Reply:
I think action from spool status will not be in sync with the action we want to use through cloud, it will confuse user.

Reply:
Agreed. We should have an help section in UI on actions to take.